headless-Checkout/client/react/headless-checkout/src/pages/save-payment-method/SavePaymentMethod.jsx

```diff
@@ -1,7 +1,217 @@
+            ) : null}
+            {shouldRenderPaymentMethodOptions ? (
+                <div className="container w-20 flex flex-column gap-10">
+                    {
+                        paymentMethodOptions.map((option, i) => (
+                            <div className="flex flex-column gap-10">
+                                <div key={`payment-method-${i}`} id={option.rail_code} className="btn btn-1" onClick={() => {
+                                    if (selectedPaymentMethod === option.rail_code) {
+                                        setSelectedPaymentMethod(null);
+                                        setPaymentDetails({});
+                                    } else {
+                                        setSelectedPaymentMethod(option.rail_code);
+                                        // create new payment details as per payment method fields
+                                        const newPaymentDetails = createInitialPaymentDetails(option.form_fields);
+                                        setPaymentDetails({...newPaymentDetails});
+                                    }
+                                }} >
+                                    {option.rail_code}
+                                </div>
+                                {(selectedPaymentMethod === option.rail_code) ? (option.form_fields.length && (
+                                    <div className="flex flex-column gap-10 my-15">
+                                        {option.form_fields.map((field, index) => (
+                                            <div key={`field-${index}`} className="field flex flex-column gap-5">
+                                                <label htmlFor={field.name}>
+                                                    <span className="font-weight-bold">{field.label}</span>
+                                                    {field.required ? (<span className="required-field-char">*</span>) : ''}
+                                                </label>
+                                                <input type={(field.field_type === 'textfield') ? 'text' : field.field_type} id={field.name} placeholder={field.placeholder} className={`input-entry ${!paymentDetails[field.name].value.length || paymentDetails[field.name].isValidated ? '' : 'invalid-entry'}`} value={paymentDetails[field.name].value} onChange={(e) => handleChange(e, field)} />
+                                            </div>
+                                        ))}
+                                    </div>
+                                )) || (
+                                    <div className="text-align-center my-15">No fields to display!</div>
+                                ) : null}
+                            </div>
+                        ))
+                    }
+                    <div className="btn btn-1 btn-bg-color-1 border-radius-1 my-3" onClick={save}>SAVE</div>
```

Comment:
lets not give styles like this for buttons too

Reply:
taking css changes separately